content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,288 @@
+
+Resource driver controller has to implement a [driver interface](https://pkg.go.dev/k8s.io/dynamic-resource-allocation@v0.28.1/controller#Driver),
+which then can be simply used with [controller helper instance](https://pkg.go.dev/k8s.io/dynamic-resource-allocation@v0.28.1/controller#New).
+
+See the [example resource driver controller](https://github.com/kubernetes-sigs/dra-example-driver/blob/151b7c8da2e620c47a3591e1a937f8d0297b0c25/cmd/dra-example-controller/main.go#L208) for details.
+
+
+### Kubelet-plugin registration
+
+Resource driver kubelet-plugin has to implement [NodeServer interface](https://pkg.go.dev/k8s.io/kubelet@v0.28.2/pkg/apis/dra/v1alpha3#NodeServer),
+which can be simply used with [kubelet-plugin helper instance](https://github.com/kubernetes-sigs/dra-example-driver/blob/151b7c8da2e620c47a3591e1a937f8d0297b0c25/cmd/dra-example-kubeletplugin/main.go#L160).
+You can use any programming language, but the examples are all written in Go.
+
+## Controller implementation
+
+A resource driver controller's main responsibility is to allocate and deallocate resources for
+{{< glossary_tooltip term_id="resource-claim" text="ResourceClaims">}}.
+
+There are two modes of allocation the ResourceClaim can have:
+
+- `WaitForFirstConsumer` (default), AKA "delayed". In this mode the resource will only be requested
+  for ResourceClaim when a Pod that needs it is being scheduled.
+- `Immediate`: the resource has to be allocated to the ResourceClaim as soon as possible, and
+  retained until the ResourceClaim is deleted.
+
+If `allocationMode` is not set explicitly, the mode is `WaitForFirstConsumer`.
+
+### Delayed allocation
+
+Delayed allocation is done in two steps: first, the scheduler picks up a list of potential Node names
+and asks from resource driver controller which of them are unsuitable for ResourceClaim through
+_UnsuitableNodes_ call. If no nodes were suitable, scheduler selects another batch of Node names for
+the same call, until suitable node is found.
+
+After at least one Node is found to be suitable for all ResourceClaims, the scheduler
+considers the suitable nodes for the other Pod scheduling constraints (native resources
+requests, affinity, selectors, etc.), and picks up exactly one Node name. The resource driver
+controller is then asked to allocate resources for the ResourceClaim on the selected node through
+_Allocate_ call. All ResourceClaims, that Pod references and that match the resource driver name in
+ResourceClass, will be passed as parameters into this call.
+
+If Allocate call returns error for any number of ResourceClaims, the helper code will repeat the
+same call with interval until it succeeds.
+
+### Immediate allocation
+
+Immediate allocation does not have selected node, and it is up to the Resource Driver controller
+to select the best suitable node based on the ResourceClaim, ResourceClass and their parameters.
+Therefore in this scenario the `UnsupportedNodes` is not called, but only `Allocate`.
```

Comment:
I'd put more emphasis on the API that's actually used, than the one that is not:
```suggestion
Therefore in this scenario only `Allocate` is called (without `UnsupportedNodes` being called first).
```
(I had trouble parsing this on first reading as my eyes skipped "not".)